src/shared-components/carousel/arrow/index.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { ArrowLeftIcon, ArrowRightIcon } from 'src/icons';
+import { COLORS } from 'src/constants';
+
+import ArrowContainer from './style';
+
+const Arrow = props => {
+  const {
+    style,
+    onClick,
+    onUserInteraction,
+    prev,
+    next,
+    disabled,
+    ...rest
+  } = props;
+
+  return (
+    <ArrowContainer
+      {...rest}
+      style={{ ...style }}
+      onClick={() => {
```

Comment:
We should probably just turn this into a class component and define an `onClick` handler for the class that is equal to this function definition so that we can avoid defining an anonymous function here.

Reply:
wow, it turns out we neednt any of those props (only onClick should be passed as part of specs) I refactored and cleanup this component. I also fixed a bug where if you click the arrow space once it disappear (opacity:0) it throws an error now I handle it with display:none